tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -194,6 +195,28 @@ public function testFetchAllColumn() : void
         self::assertEquals([1], $stmt->fetchAll(FetchMode::COLUMN));
     }
 
+    public function testGetIterator() : void
+    {
+        $query = $this->connection->getDatabasePlatform()
+            ->getDummySelectSQL('1');
+
+        $qcp = new QueryCacheProfile(0, 0, new ArrayCache());
+
+        $stmt = $this->connection->executeCacheQuery($query, [], [], $qcp);
+        $stmt->/** @scrutinizer ignore-call */
+                getIterator();
+        $stmt->closeCursor();
+
+        $stmt = $this->connection->executeCacheQuery($query, [], [], $qcp);
+
+        $iterator = $stmt->/** @scrutinizer ignore-call */
+                            getIterator();
```

Comment:
What did Scrutinizer have to say about this? There are currently no usages of such an annotation in the codebase, after all these years of development, hence my question.

Reply:
The Scrutinizer said that don't have getIterator in ResultStatement in this is true, but here we call ResultCacheStatement indeed

Reply:
Scrutinizer is not wrong, sometimes it returns something else:

https://github.com/doctrine/dbal/blob/4c03ed81471c62f178581eb325339a3c34f3b71a/lib/Doctrine/DBAL/Connection.php#L954-L965

The thing is, `ArrayStatement` also has a `getIterator` method. Try changing this line:
https://github.com/doctrine/dbal/blob/4c03ed81471c62f178581eb325339a3c34f3b71a/lib/Doctrine/DBAL/Connection.php#L934

Use `ResultCacheStatement|ArrayStatement`, and you should be able to remove these annotations.